urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
Let's avoid the scope creep here and restrict ourselves to the specific fix in question here. A separate issue to restructure the exception heirarchy is a good idea though.